services/contract-call-decoder/src/index.ts

```diff
@@ -37,4 +52,146 @@ export default class ContractCallDecoder {
       }
       throw new Error(`Couldn't decode the hash format: ${origin}`);
   }
+  
+  /**
+   * Function to decode a human readable documentation for the called function.
+   * 
+   * @param hexTxInput - input transaction data in hex string
+   * @param metadataOutput - output field of the metadata.json. Includes abi, userdoc, devdoc
+   * @returns an Object with all extractable useful information
+   * 
+   * @example
+   * return {
+      functionName: 'mint',
+      params: [
+        {
+          name: '_to',
+          type: 'address',
+          value: '0xAA6042aa65eb93C6439cDaeBC27B3bd09c5DFe94'
+        },
+        { name: '_amount', type: 'uint256', value: '1000000000000000000' }
+      ],
+      userdoc: {
+        notice: 'Creates 1000000000000000000 token to 0xAA6042aa65eb93C6439cDaeBC27B3bd09c5DFe94. Must only be called by the owner (MasterChef).'
+      },
+      devdoc: undefined
+    }
+   */
```

Comment:
I do rather comment extensively than too few to allow a reader to easily understand what the code does. In particular, since this will be an npm package, this will allow users to hover over the function and see what it does, including an example input and output. I find packages with embedded comments really useful. You can `Ctrl + Shift + P` then `Fold All Block Comments` in VSCode for clean view. Did these comments not help much what it does when you were reviewing? Lmk

Reply:
I will go ahead to approve either ways.